utils/filesystem/zipfs_test.go

```diff
@@ -0,0 +1,264 @@
+	require.NoError(t, err)
+	defer func() {
+		if zipFile != nil {
+			_ = zipFile.Close()
+		}
+	}()
+	require.NotNil(t, zipFile)
+	assert.False(t, fs.Exists(faker.DomainName()))
+	// FIXME: enable when issue in afero is fixed https://github.com/spf13/afero/issues/395
+	// assert.True(t, fs.Exists(string(filepath.Separator)))
+	// assert.True(t, fs.Exists("/"))
+	assert.True(t, fs.Exists("testunzip/test.txt"))
+	assert.True(t, fs.Exists("testunzip/child.zip"))
+	assert.True(t, fs.Exists("testunzip/ไป ไหน มา.txt"))
+	require.NoError(t, zipFile.Close())
+}
+
+func Test_zipFs_Exists2(t *testing.T) {
+	// using afero test zip file
+	fs, zipFile, err := NewZipFileSystemFromStandardFileSystem(filepath.Join("testdata", "t.zip"), NoLimits())
+	require.NoError(t, err)
+	defer func() {
+		if zipFile != nil {
+			_ = zipFile.Close()
+		}
+	}()
+	require.NotNil(t, zipFile)
+	assert.False(t, fs.Exists(faker.DomainName()))
+	assert.True(t, fs.Exists(string(filepath.Separator)))
+	assert.True(t, fs.Exists("/"))
+
+	assert.True(t, fs.Exists("testDir1"))
+	assert.True(t, fs.Exists("testFile"))
+	assert.True(t, fs.Exists("testDir1/testFile"))
+	require.NoError(t, zipFile.Close())
+}
+
+func Test_zipFS_FileInfo(t *testing.T) {
+	fs, zipFile, err := NewZipFileSystemFromStandardFileSystem(filepath.Join("testdata", "testunzip.zip"), NoLimits())
+	require.NoError(t, err)
+	defer func() {
+		if zipFile != nil {
+			_ = zipFile.Close()
+		}
+	}()
+	require.NotNil(t, zipFile)
+	zfile, err := fs.Stat("/")
+	require.NoError(t, err)
+	assert.Equal(t, string(filepath.Separator), zfile.Name())
+	assert.True(t, zfile.IsDir())
+	assert.Zero(t, zfile.Size())
+
+	zfile, err = fs.Stat("testunzip/test.txt")
+	require.NoError(t, err)
+	assert.Equal(t, "test.txt", zfile.Name())
+	assert.False(t, zfile.IsDir())
+	assert.NotZero(t, zfile.Size())
+
+	require.NoError(t, zipFile.Close())
+}
+
+func Test_zipFs_Browsing(t *testing.T) {
+	fs, zipFile, err := NewZipFileSystemFromStandardFileSystem(filepath.Join("testdata", "testunzip.zip"), NoLimits())
+	require.NoError(t, err)
+	defer func() {
+		if zipFile != nil {
+			_ = zipFile.Close()
+		}
+	}()
+	require.NotNil(t, zipFile)
+	empty, err := fs.IsEmpty(faker.DomainName())
+	require.NoError(t, err)
+	assert.True(t, empty)
+	empty, err = fs.IsEmpty("testunzip/test.txt")
+	require.NoError(t, err)
+	assert.False(t, empty)
+	empty, err = fs.IsEmpty("testunzip/child.zip")
+	require.NoError(t, err)
+	assert.False(t, empty)
+	empty, err = fs.IsEmpty("testunzip/ไป ไหน มา.txt")
+	require.NoError(t, err)
+	assert.True(t, empty)
+	require.NoError(t, zipFile.Close())
+}
+
+func Test_zipFs_Browsing2(t *testing.T) {
+	zipFs, zipFile, err := NewZipFileSystemFromStandardFileSystem(filepath.Join("testdata", "t.zip"), NoLimits())
+	require.NoError(t, err)
+	defer func() {
+		if zipFile != nil {
+			_ = zipFile.Close()
+		}
+	}()
+	require.NotNil(t, zipFile)
+	var wFunc = func(path string, info fs.FileInfo, err error) error {
```

Comment:
could we pre generate a list of the files using ListDir and then walk it and make sure they match instead of printing

Reply:
will do